src/LoggingConnector.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace LeProxy\LeProxy;
+
+use React\Socket\Connector;
+use React\Promise\Promise;
+use React\Stream\WritableStreamInterface;
+use React\Socket\ConnectorInterface;
+use React\Socket\ConnectionInterface;
+use Clue\React\HttpProxy\ProxyConnector as HttpClient;
+use Clue\React\Socks\Client as SocksClient;
+
+/** @internal */
+class LoggingConnector implements ConnectorInterface
+{
+    private $connector;
+    private $outputStream;
+    private $protocol = 'tcp';
+    private $auth = null;
+
+    public function __construct(ConnectorInterface $connector, WritableStreamInterface $outputStream)
+    {
+        if ($connector instanceof HttpClient) {
+            $this->protocol = 'http';
+        }
+
+        if ($connector instanceof SocksClient) {
+            $this->protocol = 'socks5';
+        }
+
+        $this->connector = $connector;
+        $this->outputStream = $outputStream;
+    }
+
+    public function connect($uri)
+    {
+        $that = $this;
+        $connector = $this->connector;
+        $auth = $this->auth;
+
+        return new Promise(
+            function ($resolve, $reject) use ($connector, $uri, $that, $auth) {
+                $connector->connect($uri)->then(
+                    function (ConnectionInterface $remote) use ($that, $uri, $resolve, $auth) {
```

Comment:
Code style is a bit unclear here, does this target PHP < 5.4, in which case `$this` must not be used or if this targets only PHP > 5.3, then most of these variable bindings are unneeded.